docs/connector-development/config-based/overview.md

```diff
@@ -0,0 +1,95 @@
+# Config-based connectors overview
+
+The goal of this document is to give enough technical specifics to understand how config-based connectors work.
+When you're ready to start building a connector, you can start with [the tutorial](../../../config-based/tutorial/0-getting-started.md) or dive into the [reference documentation](https://airbyte-cdk.readthedocs.io/en/latest/api/airbyte_cdk.sources.declarative.html)
+
+## Overview
+
+Config-based connectors work by parsing a YAML configuration describing the Source, then running the configured connector using a Python backend.
+
+The process then submits HTTP requests to the API endpoint, and extracts records out of the response.
+
+See the [connector definition section](connector-definition.md) for more information on the YAML file describing the connector.
+
+## Source
+
+Config-based connectors are a declarative way to define HTTP API sources.
+
+A source is defined by 2 components:
+
+1. The source's `Stream`s, which define the data to read
+2. A `ConnectionChecker`, which describes how to run the `check` operation to test the connection to the API source
+
+## Stream
+
+Streams define the schema of the data to sync, as well as how to read it from the underlying API source.
+A stream generally corresponds to a resource within the API. They are analogous to tables for a RDMS source.
+
+A stream is defined by:
+
+1. A name
+2. A primary key: used to uniquely identify records, enabling deduplication
+3. A schema: describes the data to sync
+4. A data retriever: describes how to retrieve the data from the API
+5. A cursor field: used to identify the stream's state from a record
+6. A set of transformations to be applied on the records read from the source before emitting them to the destination
+7. A checkpoint interval: defines when to checkpoint syncs.
+
+More details on streams and sources can be found in the [basic concepts section](../cdk-python/basic-concepts.md).
+More details on cursor fields, and checkpointing can be found in the [incremental-stream section](../cdk-python/incremental-stream.md)
+
+## Data retriever
+
+The data retriever defines how to read the data from an API source, and acts as an orchestrator for the data retrieval flow.
+There is currently only one implementation, the `SimpleRetriever`, which is defined by
+
```

Comment:
Could you explain/remind if the retriever is stream specific, global to a source, or can be both?

Reply:
updated